/*
 * Copyright (C) Mellanox Technologies Ltd. 2001-2013.  ALL RIGHTS RESERVED.
 *
 * This software product is a proprietary product of Mellanox Technologies Ltd.
 * (the "Company") and all right, title, and interest in and to the software product,
 * including all associated intellectual property rights, are and shall
 * remain exclusively with the Company.
 *
 * This software is made available under either the GPL v2 license or a commercial license.
 * If you wish to obtain a commercial license, please contact Mellanox at support@mellanox.com.
 */


This patch peforms the following changes in pcb:
* change snd_wnd filed dt from u16_t to u32_t.
* adds max_snd_buffs, max_unsent_len fields which allows dynamic change of the max send window size

--- lwip.orig/src/include/lwip/tcp.h	2010-07-16 17:10:57.000000000 +0300
+++ lwip/src/include/lwip/tcp.h	2011-06-05 13:10:52.636719000 +0300
@@ -220,14 +220,16 @@
 
   /* sender variables */
   u32_t snd_nxt;   /* next new seqno to be sent */
-  u16_t snd_wnd;   /* sender window */
+  u32_t snd_wnd;   /* sender window */
   u32_t snd_wl1, snd_wl2; /* Sequence and acknowledgement numbers of last
                              window update. */
   u32_t snd_lbb;       /* Sequence number of next byte to be buffered. */
 
   u16_t acked;
   
-  u16_t snd_buf;   /* Available buffer space for sending (in bytes). */
+  u32_t snd_buf;   /* Available buffer space for sending (in bytes). */
+  u32_t max_snd_buff;
+
 #define TCP_SNDQUEUELEN_OVERFLOW (0xffff-3)
   u16_t snd_queuelen; /* Available buffer space for sending (in tcp_segs). */
 
@@ -235,9 +237,10 @@
   /* Extra bytes available at the end of the last pbuf in unsent. */
   u16_t unsent_oversize;
 #endif /* TCP_OVERSIZE */ 
-
+  u16_t max_unsent_len;
   /* These are ordered by sequence number: */
   struct tcp_seg *unsent;   /* Unsent (queued) segments. */
+  struct tcp_seg *last_unsent;   /* Last unsent (queued) segment. */
   struct tcp_seg *unacked;  /* Sent but unacknowledged segments. */
 #if TCP_QUEUE_OOSEQ  
   struct tcp_seg *ooseq;    /* Received out of sequence segments. */
